scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```
@@ -0,0 +1,399 @@
+      try {
+        processor.accept(metadata, dto, timeZone);
+      } catch (XPathException e) {
+        log.error("Failed to extract metadata", e);
+      }
+    }
+  }
+
+  /**
+   * Tests whether a String is blank (empty or just spaces) or null. Duplicated from MISO's
+   * LimsUtils.
+   *
+   * @param s String to test for blank or null
+   * @return true if blank or null String provided
+   */
+  private static boolean isStringBlankOrNull(String s) {
+    return s == null || s.trim().isEmpty();
+  }
+
+  @Override
+  public PathType getPathType() {
+    return PathType.DIRECTORY;
+  }
+
+  /**
+   * Returns a stream that includes Metadata directory level to be used for further processing
+   *
+   * @param runDirectory which we are currently processing
+   * @param directoryName specific sub-directory we want
+   * @return Stream from Metadata directory level
+   */
+  private Stream<File> streamSmrtCellSubdirectories(File runDirectory, String directoryName) {
+    return Stream.of(runDirectory.listFiles())
+        .filter(
+            cellDirectory ->
+                cellDirectory.isDirectory() && REVIO_CELL_DIRECTORY.test(cellDirectory.getName()))
+        .flatMap(
+            cellDirectory ->
+                Stream.of(cellDirectory.listFiles())
+                    .filter(
+                        subDirectory ->
+                            subDirectory.isDirectory()
+                                && subDirectory.getName().equals(directoryName)));
+  }
+
+  /**
+   * Returns the time a file was created
+   *
+   * @param file which we are currently processing
+   * @return Instant point in time
+   */
+  private static Instant getFileCreationTime(File file) {
+    try {
+      FileTime fileTime =
+          (FileTime) Files.getAttribute(Path.of(file.getAbsolutePath()), "creationTime");
+      return fileTime.toInstant();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
```

Change this method to throw `IOException` instead of catching it. The `process` method is already expected to throw it, so it's fine to let this bubble up

---

Missed this one

---

The easiest good solution would be to do this outside of a stream